### PR TITLE
Improved MTurk credential override UX

### PR DESCRIPTION
**Pull request by Yash621**
@JackUrb I have done the updations that you suggested ,please review it !
I have also tested it and it is working fine.
This PR is in respect to issue #531 and a continuation of PR #615

### CODE REVIEW

**Review by JackUrb — APPROVED** [review]
Alright this is looking good. Some small nits still inline, but I really appreciate that you've stuck with this set of changes and brought it across the finish line!

File: mephisto/abstractions/providers/mturk/mturk_utils.py [hunk 1]
```diff
@@ -61,21 +61,40 @@ def check_aws_credentials(profile_name: str) -> bool:
...
+            profileFound = False
+            for credentialIndex in range(0, len(aws_credentials)):
+                if profileFound:
+                    aws_credentials[credentialIndex] = "aws_access_key_id={}\n".format(
+                        register_args.access_key_id)
+                    aws_credentials[credentialIndex+1] = "aws_access_key_id={}\n".format(
+                        register_args.access_key_id)
+                    break
+                if str(aws_credentials[credentialIndex]).startswith("[{}]".format(profile_name)):
+                    profileFound = True
```
Comment (JackUrb) [inline mephisto/abstractions/providers/mturk/mturk_utils.py:89]: This flag iteration style loop is still somewhat opaque - there's no need to take the extra iteration step once you've found the proflie:
```
for credentialIndex in range(0, len(aws_credentials)):
     if str(aws_credentials[credentialIndex]).startswith("[{}]".format(profile_name)):
        aws_credentials[credentialIndex+1] = "aws_access_key_id={}\n".format(register_args.access_key_id)
        aws_credentials[credentialIndex+2] = "aws_access_key_id={}\n".format(register_args.access_key_id)
        break
```

File: mephisto/abstractions/providers/mturk/mturk_utils.py [hunk 2]
```diff
@@ -61,21 +61,40 @@ def check_aws_credentials(profile_name: str) -> bool:
...
+
+                with open(expanded_aws_file_path, "w") as aws_credentials_file:
+                    # overWrite login details
+                    aws_credentials_file.write("\n".join(aws_credentials))
+                logger.warning(f"The provided keys does not match for profile {profile_name}"
+                               f"As a result of which we're updating the credentials, overwriting ones that already existed for the profile ")
```
Comment (JackUrb) [inline mephisto/abstractions/providers/mturk/mturk_utils.py:95]: Note, we now override always, and aren't checking to see if we've actually made any changes. A valid message would then be:
```
logger.warning(
    f"We found an existing entry for {profile_name}. As new credentials have been provided, "
    f"we're updating the credentials, overwriting ones that already existed for the profile "
)
```

**Comment by Yash621** [comment]
@JackUrb thanks and I look forward to learn a lot from you !

**Comment by Yash621** [comment]
@JackUrb I have done those small changes also that you suggested ,please see if anything else is needed.

**Review by JackUrb — APPROVED** [review]

**Comment by JackUrb** [comment]
Seems code style is failing - you'll want to install `black` as that's what we use for automated formatting. You can install it as part of our precommit hook, running `pip install pre-commit`. Sorry I should have noticed this sooner!

**Comment by Yash621** [comment]
@JackUrb no problem ! I have updated the code formatting.

**Comment by JackUrb** [comment]
Thanks for taking this on! I can manage the rest of the cleanup in any next merge. I'm happy to help in the future if you want to contribute again!

**Comment by Yash621** [comment]
sure ! thanks for your guidance and support :)

**Pull request merged**